### PR TITLE
refactor: Move units check outside of model for finds

### PR DESCRIPTION
**Pull request by KevLehman**
OC-950

## Further comments

- Removes Units check from inside `find` and `findpaginated` functions.
- Since updates are async, having the extra filter inside is not a problem.
- Removes `Promise.await` introduced to fix a regression on 6.2

### CODE REVIEW

File: apps/meteor/ee/server/models/raw/LivechatRooms.ts [hunk 1]
```diff
@@ -159,6 +168,7 @@ export class LivechatRoomsRawEE extends LivechatRoomsRaw implements ILivechatRoo
 		);
 	}
 
+	// TODO - check if removable, looks unused
```
Comment (murtaza98) [inline apps/meteor/ee/server/models/raw/LivechatRooms.ts:171]: I think you're right here. This indeed doesn't seem to be used anywhere. Perhaps we can remove it?

File: apps/meteor/ee/app/livechat-enterprise/server/lib/query.helper.js [hunk 2]
```diff
@@ -18,13 +18,18 @@ export const addQueryRestrictionsToRoomsModel = (originalQuery = {}) => {
 	return query;
 };
 
-export const addQueryRestrictionsToDepartmentsModel = async (originalQuery = {}) => {
-	const query = { ...originalQuery, type: { $ne: 'u' } };
+export const restrictQuery = async (originalQuery = {}) => {
```
Comment (murtaza98) [inline apps/meteor/ee/app/livechat-enterprise/server/lib/query.helper.js:21]: Is this still needed, considering you've already moved this logic to `applyDepartmentRestrictions.ts`

**Review by murtaza98 — APPROVED** [review]

**Review by murtaza98 — DISMISSED** [review]

**Review by d-gubert — APPROVED** [review]
All good on apps side

**Review by ggazzo — APPROVED** [review]

**Review by d-gubert — APPROVED** [review]
Still good on apps side

**Review by aleksandernsilva — APPROVED** [review]

**Pull request merged**